modules/effects/spec/effect_creator.spec.ts

```diff
@@ -30,6 +30,14 @@ describe('createEffect()', () => {
     expect(effect['__@ngrx/effects_create__']).toEqual({ dispatch: false });
   });
 
+  it('should be possible to create a non-dispatching effect returning a non-action', () => {
+    const effect: any = createEffect(() => of('foo'), {
+      dispatch: false,
+    });
+
+    expect(effect['__@ngrx/effects_create__']).toEqual({ dispatch: false });
+  });
+
```

Comment:
Could you please add a test with `expectSnippet` to catch that `createEffect(() => of('foo'))` _/is not assignable to type 'Observable<Action>/_ ?
I think it could be beneficial - right not nothing prevents from accidentally adjusting the code to return `Observable<any>`